frontend/src/metabase/query_builder/components/notebook/steps/JoinStep.jsx

```diff
@@ -547,7 +547,7 @@ function getDimensionDisplayName(dimension) {
   if (!dimension) {
     return t`Pick a column...`;
   }
-  if (dimension.temporalUnit()) {
+  if (isDateTimeField(dimension.mbql())) {
```

Comment:
`temporalUnit` method does not exist on `ExpressionDimension` instances, so that's why it fails. Fixed by checking MBQL instead